pulpcore/pulpcore/app/tasks/importer.py

```diff
@@ -80,4 +88,8 @@ def sync(importer_pk):
                 'repository': importer.repository.name,
                 'importer': importer.name
             })
-        importer.sync()
+        importer.sync(new_version, old_version)
```

Comment:
Suggest:
```
try:
    importer.sync(new_version, old_version)
except Exception:
    new_version.delete()
```
Another thing to consider is that when the new version is saved without transactions, it will be visable in a *half-baked* state for the duration of the sync.  This might be bad for read operations not locked by reservation like applicability calculation.